app/client/components/delivery/records/deliveryRecords.tsx

```diff
@@ -1,77 +1,591 @@
+    boolean
+  >(false);
+  const [step2FormValidationDetails, setStep2FormValidationDetails] = useState<
+    Step1FormValidationDetails
+  >({ isValid: true });
+  const [deliveryProblem, setDeliveryProblem] = useState<
+    DeliveryRecordsProblemType
+  >();
+  const [
+    showTopCallCentreNumbers,
+    setTopCallCentreNumbersVisibility
+  ] = useState<boolean>(false);
+  const [
+    showBottomCallCentreNumbers,
+    setBottomCallCentreNumbersVisibility
+  ] = useState<boolean>(false);
+  const step1FormUpdateCallback = (isValid: boolean, message?: string) => {
+    setStep1formValidationState(false);
+    setStep1FormValidationDetails({ isValid, message });
+  };
+  const step1FormSubmitListener = (
+    selectedValue: string | undefined,
+    selectedMessage: string | undefined
+  ) => {
+    setDeliveryProblem({ category: selectedValue, message: selectedMessage });
+    setStep1formValidationState(true);
+    if (step1FormValidationDetails.isValid) {
+      trackEvent({
+        eventCategory: "delivery-problem",
+        eventAction: "continue_to_step_2_button_click",
+        product: {
+          productType: props.routeableStepProps.productType,
+          productDetail: props.productDetail
+        },
+        eventLabel: props.routeableStepProps.productType.urlPart
+      });
+      setPageStatus(PageStatus.REPORT_ISSUE_STEP_2);
+    }
+  };
+
+  const addRecordToDeliveryProblem = (id: string) =>
+    setSelectedProblemRecords([...selectedProblemRecords, id]);
+  const removeRecordFromDeliveryProblem = (id: string) =>
+    setSelectedProblemRecords(
+      selectedProblemRecords.filter((existingId: string) => existingId !== id)
+    );
+  const resultsPerPage = 7;
+  const totalPages = Math.ceil(props.data.results.length / resultsPerPage);
+  const scrollToTop = () => window.scrollTo(0, 0);
+  const resetDeliveryRecordsPage = () => setPageStatus(PageStatus.READ_ONLY);
+
+  const filterData = (productPartName: ProductUrlPart) => {
+    const NumOfRecordsToShow =
+      ProductTypes[productPartName as ProductTypeKeys].delivery?.records
+        ?.numberOfProblemRecordsToShow || props.data.results.length;
+
+    if (pageStatus !== PageStatus.READ_ONLY) {
+      return props.data.results
+        .filter(record => !record.problemCaseId)
+        .slice(0, NumOfRecordsToShow - 1);
+    }
+    return props.data.results.filter((element, index) =>
+      isRecordInCurrentPage(
+        index,
+        currentPage * resultsPerPage,
+        currentPage * resultsPerPage + resultsPerPage - 1
+      )
+    );
+  };
+
+  const capatalize = (input: string) =>
+    input.charAt(0).toUpperCase() + input.slice(1);
+
+  const isRecordInCurrentPage = (
+    index: number,
+    currentPageStartIndex: number,
+    currentPageEndIndex: number
+  ) => index >= currentPageStartIndex && index <= currentPageEndIndex;
+
+  const hasExistingDeliveryProblem =
+    props.data.results.findIndex(
+      deliveryRecord => deliveryRecord.problemCaseId
+    ) > -1;
```

Comment:
i think we need to change this implementation to make it explicitly last 14 days...
```suggestion
  const hasExistingDeliveryProblem =
    props.data.results.findIndex(
      deliveryRecord => (deliveryRecord.problemCaseId && deliveryRecord.date.minusDays(14) > TODAY)
    ) > -1;
```

Reply:
Just realised this suggestion is wrong... it should be more like ```deliveryRecord.deliveryDate > TODAY.minusDays(15)```

Reply:
If it's using Moments for the date substraction, make sure to clone() if referring to an existing variable for 'TODAY' otherwise it mutates the moment 😱